### PR TITLE
Test helper improvements

### PR DESCRIPTION
**Pull request by st0012**
### Motivation

1. We should pass file and line number when calling `class_eval` so the methods generated inside can get proper location for debugging.
2. The `diff` method's logic is incorrect so it never prints `expected` and `actual` value in JSON format. This makes copy and pasting the `actual` value to new `.exp` file impossible. See the 2nd commit's message for more information.

**Before**

```
DocumentLinkExpectationsTest
########## Expected ##########
{"range"=>{"start"=>{"line"=>0, "character"=>0}, "end"=>{"line"=>0, "character"=>50}}, "target"=>"file:///Users/hung-wulo/.gem/ruby/3.1.2/gems/syntax_tree-3.5.0/lib/syntax_tree.rb#39", "tooltip"=>"Jump to /Users/hung-wulo/.gem/ruby/3.1.2/gems/syntax_tree-3.5.0/lib/syntax_tree.rb#39"}
{"range"=>{"start"=>{"line"=>4, "character"=>0}, "end"=>{"line"=>4, "character"=>32}}, "target"=>"file:///opt/rubies/3.1.2/lib/ruby/3.1.0/mutex_m.rb#1", "tooltip"=>"Jump to /opt/rubies/3.1.2/lib/ruby/3.1.0/mutex_m.rb#1"}
##########  Actual  ##########
{"range"=>{"start"=>{"line"=>0, "character"=>0}, "end"=>{"line"=>0, "character"=>50}}, "target"=>"file:///Users/hung-wulo/.gem/ruby/3.1.2/gems/syntax_tree-3.5.0/lib/syntax_tree.rb#39", "tooltip"=>"Jump to /Users/hung-wulo/.gem/ruby/3.1.2/gems/syntax_tree-3.5.0/lib/syntax_tree.rb#39"}
{"range"=>{"start"=>{"line"=>4, "character"=>0}, "end"=>{"line"=>4, "character"=>32}}, "target"=>"file:///opt/rubies/3.1.2/lib/ruby/3.1.0/mutex_m.rb#1", "tooltip"=>"Jump to /opt/rubies/3.1.2/lib/ruby/3.1.0/mutex_m.rb#1"}
{"range"=>{"start"=>{"line"=>8, "character"=>0}, "end"=>{"line"=>8, "character"=>45}}, "target"=>"file:///Users/hung-wulo/.gem/ruby/3.1.2/gems/syntax_tree-3.5.0/lib/syntax_tree.rb#39", "tooltip"=>"Jump to /Users/hung-wulo/.gem/ruby/3.1.2/gems/syntax_tree-3.5.0/lib/syntax_tree.rb#39"}
```

**After**

```
########## Expected ##########
{
  "result": [
    {
      "range": {
        "start": {
          "line": 0,
          "character": 0
        },
        "end": {
          "line": 0,
          "character": 50
        }
      },
      "target": "file:///Users/hung-wulo/.gem/ruby/3.1.2/gems/syntax_tree-3.5.0/lib/syntax_tree.rb#39",
      "tooltip": "Jump to /Users/hung-wulo/.gem/ruby/3.1.2/gems/syntax_tree-3.5.0/lib/syntax_tree.rb#39"
    },
    {
      "range": {
        "start": {
          "line": 4,
          "character": 0
        },
        "end": {
          "line": 4,
          "character": 32
        }
      },
      "target": "file:///opt/rubies/3.1.2/lib/ruby/3.1.0/mutex_m.rb#1",
      "tooltip": "Jump to /opt/rubies/3.1.2/lib/ruby/3.1.0/mutex_m.rb#1"
    }
  ],
  "params": [

  ]
}
##########  Actual  ##########
{
  "result": [
    {
      "range": {
        "start": {
          "line": 0,
          "character": 0
        },
        "end": {
          "line": 0,
          "character": 50
        }
      },
      "target": "file:///Users/hung-wulo/.gem/ruby/3.1.2/gems/syntax_tree-3.5.0/lib/syntax_tree.rb#39",
      "tooltip": "Jump to /Users/hung-wulo/.gem/ruby/3.1.2/gems/syntax_tree-3.5.0/lib/syntax_tree.rb#39"
    },
    {
      "range": {
        "start": {
          "line": 4,
          "character": 0
        },
        "end": {
          "line": 4,
          "character": 32
        }
      },
      "target": "file:///opt/rubies/3.1.2/lib/ruby/3.1.0/mutex_m.rb#1",
      "tooltip": "Jump to /opt/rubies/3.1.2/lib/ruby/3.1.0/mutex_m.rb#1"
    },
    {
      "range": {
        "start": {
          "line": 8,
          "character": 0
        },
        "end": {
          "line": 8,
          "character": 45
        }
      },
      "target": "file:///Users/hung-wulo/.gem/ruby/3.1.2/gems/syntax_tree-3.5.0/lib/syntax_tree.rb#39",
      "tooltip": "Jump to /Users/hung-wulo/.gem/ruby/3.1.2/gems/syntax_tree-3.5.0/lib/syntax_tree.rb#39"
    }
  ],
  "params": [

  ]
}
```

### Implementation

See the commit messages.

### Automated Tests

All the existing tests should provide enough coverage.

### Manual Tests

1. Make an expectation test fail by changing its content (but keep it a valid json)
2. Run the test
3. The expected/actual output should be json instead of Ruby Hash

### CODE REVIEW

File: test/expectations/expectations_test_runner.rb [hunk 1]
```diff
@@ -42,13 +42,13 @@ def default_args
 
         required_ruby_version = ruby_requirement_magic_comment_version(path)
         if required_ruby_version && RUBY_VERSION < required_ruby_version
-          class_eval(<<~RB)
+          class_eval(<<~RB, __FILE__, __LINE__ + 1)
```
Comment (egiurleo) [inline test/expectations/expectations_test_runner.rb:45]: This is so cool and I had no idea you could do that!

Comment (egiurleo) [reply test/expectations/expectations_test_runner.rb:45]: Is there documentation anywhere that talks about this or do you just know about it from experience?

**Review by egiurleo — APPROVED** [review]

**Review by dirceu — APPROVED** [review]
<3

File: test/expectations/expectations_test_runner.rb (hunk 1 above)
Comment (st0012) [reply test/expectations/expectations_test_runner.rb:45]: Many blog posts mentioned it but it looks like the [source code comment doesn't provide any example](https://rubyapi.org/3.1/o/module#method-i-class_eval). It feels like an opportunity to contribute!

**Pull request merged**